core/src/main/java/org/apache/iceberg/rest/ExponentialHttpRequestRetryStrategy.java

```diff
@@ -148,4 +179,14 @@ public TimeValue getRetryInterval(HttpResponse response, int execCount, HttpCont
 
     return TimeValue.ofMilliseconds(delayMillis + jitter);
   }
+
+  private boolean shouldRetryIdempotent(HttpRequest request, int responseCode) {
+    if (request == null) {
+      return false;
+    }
+
+    // Check if the request is idempotent
+    return Method.isIdempotent(request.getMethod())
+        && idempotentRetriableCodes.contains(responseCode);
```

Comment:
[doubt] why these codes only is this from the rest-spec perspective ? if yes should we also add these in the above retryReq (we just check idempotency there)

Reply:
My thought was in the above context we haven't gotten any response so we have no clue what we can retry.

In this method we actually have a response from the server (or something) so there are many responses we could have gotten that are pointless to retry (not found, not authorized, etc ...) so I just added a bunch that indicate some sort of server or network based issues.

Reply:
Make sense, indeed without response code this the best shot we have :) !